cmd/vault-server/startcmd/start.go

```diff
@@ -70,23 +86,66 @@ func getParameters(cmd *cobra.Command) (*serviceParameters, error) {
 		return nil, err
 	}
 
+	remoteKMSURL, err := cmdutils.GetUserSetVarFromString(cmd, remoteKMSURLFlagName, remoteKMSURLEnvKey, false)
+	if err != nil {
+		return nil, err
+	}
+
+	edvURL, err := cmdutils.GetUserSetVarFromString(cmd, edvURLFlagName, edvURLEnvKey, false)
+	if err != nil {
+		return nil, err
+	}
+
 	return &serviceParameters{
-		host: host,
+		host:         host,
+		remoteKMSURL: remoteKMSURL,
+		edvURL:       edvURL,
 	}, err
 }
 
 func createFlags(cmd *cobra.Command) {
 	cmd.Flags().StringP(hostURLFlagName, hostURLFlagShorthand, "", hostURLFlagUsage)
+	cmd.Flags().StringP(remoteKMSURLFlagName, "", "", remoteKMSURLFlagUsage)
+	cmd.Flags().StringP(edvURLFlagName, "", "", edvURLFlagUsage)
+}
+
+const (
+	keystorePrimaryKeyURI = "local-lock://keystorekms"
+)
+
+type kmsProvider struct {
+	storageProvider storage.Provider
+	secretLock      secretlock.Service
+}
+
+func (k kmsProvider) StorageProvider() storage.Provider {
+	return k.storageProvider
+}
+
+func (k kmsProvider) SecretLock() secretlock.Service {
+	return k.secretLock
 }
 
 func startService(params *serviceParameters, srv server) error {
-	router := mux.NewRouter()
+	keyManager, err := localkms.New(keystorePrimaryKeyURI, &kmsProvider{
+		storageProvider: mem.NewProvider(),
```

Comment:
need storage to be configurable

Reply:
yeah, I know. It will be for sure. Left a TODO for now.